GitHubComments.ps1

```diff
@@ -0,0 +1,576 @@
+        html - Return HTML rendered from the body's markdown. Response will include body_html.
+        full - Return raw, text and HTML representations. Response will include body, body_text, and body_html.
+
+    .PARAMETER MediaTypeVersion
+        The version of the response to return. The default version of the API may change in the future.
+        If you're building an application and care about the stability of the API, be sure to request 
+        a specific version. More info at https://developer.github.com/v3/media/.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        New-GitHubComment -OwnerName Powershell -RepositoryName PowerShellForGitHub -IssueNumber 1 -Body "Testing this API"
+
+        Creates a new Github comment in an issue for the PowerShell\PowerShellForGitHub project.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Uri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory)]
+        [string] $IssueNumber,
+
+        [Parameter(Mandatory)]
+        [string] $Body,
+
+        [ValidateSet('raw', 'text', 'html', 'full')]
+        [string] $MediaType ='raw',
+
+        [string] $MediaTypeVersion = 'VERSION',
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+        'IssueNumber' =  (Get-PiiSafeString -PlainText $IssueNumber)
+    }
+
+    $hashBody = @{
+        'body' = $Body
+    }
+
+    $params = @{
+        'UriFragment' = "repos/$OwnerName/$RepositoryName/issues/$IssueNumber/comments"
+        'Body' = ($hashBody | ConvertTo-Json)
```

Comment:
For consistency with recent changes, please make this: `(ConvertTo-Json -InputObject $hashBody)`